### PR TITLE
User entities migration and  remove unnecessary files

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added user, student, teacher entities
-  Migrate the entity files
- Removed unnecessary files/ folders

## Associated Ticket Link(s)

- https://trello.com/c/5ggJFONW

## Related Pull Request(s)

- N/A

### CODE REVIEW

**Comment by ragib-hossain-sazim** [comment]
Why are you removing stuff here as well? Don't you already have a separate PR for it?

**Comment by faysal-alam-sazim** [comment]
> Why are you removing stuff here as well? Don't you already have a separate PR for it?

I kept auth  & users modules and the entity files that used in auth module. In this PR, I created new entities and removed previous ones. That causing many errors in auth & users module. So I've to remove unnecessary files in this PR too.

File: src/common/entities/teachers.entity.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Entity, OneToOne, PrimaryKey, Property, Rel } from "@mikro-orm/core";
+
+import { CustomBaseEntity } from "./custom-base.entity";
+import { User } from "./users.entity";
+
+@Entity({ tableName: "teachers" })
```
Comment (ishmam-iqbal-sazim) [inline src/common/entities/teachers.entity.ts:6]: Each entity should have a repository and repository type.

Please go through the template for the syntax.

File: src/users/users.dtos.ts [hunk 2]
```diff
@@ -1,31 +1,6 @@
...
-  @MaxLength(255)
-  firstName!: string;
-
-  @IsString()
-  @MinLength(2)
-  @MaxLength(255)
-  lastName!: string;
-}
 
 export class RegisterUserDto implements Pick<User, "email" | "password"> {
```
Comment (ishmam-iqbal-sazim) [inline src/users/users.dtos.ts:5]: Should remove these as well if you are not using it.

File: src/auth/auth.helpers.ts
Comment (ishmam-iqbal-sazim) [inline src/auth/auth.helpers.ts:1]: If you don't need it, remove the file. Best not to comment out the contents of the file

**Review by ishmam-iqbal-sazim — APPROVED** [review]

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**